scene/gui/flow_container.cpp

```diff
@@ -0,0 +1,249 @@
+/*************************************************************************/
+/*  flow_container.cpp                                                   */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2021 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2021 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "flow_container.h"
+
+void FlowContainer::_resort() {
+	int separation_horizontal = get_theme_constant(SNAME("hseparation"));
+	int separation_vertical = get_theme_constant(SNAME("vseparation"));
+
+	bool rtl = is_layout_rtl();
+
+	Vector<Rect2> child_rects;
+
+	// Stores the number of children per line and the calculated line height.
+	Vector<Vector2i> lines_data;
+
+	Vector2 ofs;
+	int line_height = 0;
```

Comment:
Same naming problem as with `cached_height`.

Reply:
This is a bit problematic since renaming it to `line_size` would be too confusing I think. (`line_size` could be mixed up with line length) In my opinion, `line_height` makes the code more understandable here once you know that a line here can be a vertical or a horizontal line.